03-debugging-w-chatgpt/debugging-w-chatgpt.md

```diff
@@ -0,0 +1,491 @@
+Respond with YES if you understand the code.
+```
+
+Chat GPT's response:
+> Yes, I understand the code. It consists of two classes: `Item` and `Clothing`. The `Item` class has methods to initialize an object, get its category, provide a string representation, and describe its condition. The `Clothing` class is a subclass of `Item` and adds an attribute for fabric in its initialization, along with a customized string representation.
+> 
+> If you have any specific questions or if there's anything else you'd like assistance with regarding this code, feel free to let me know!
+
+Our Prompt:
+```
+Please enumerate the lines of the code I shared
+```
+
+Chat GPT's response:
+> Certainly! Here are the enumerated lines of the code you shared:
+>
+> **Code Block 1: Item Class**
+> ```python
+>  1. import uuid
+>  2. 
+>  3. class Item:
+>  4.     def __init__(self, condition=0, id=None):
+>  5.         self.condition = condition
+>  6.         if id is None:
+>  7.             self.id = uuid.uuid4().int
+>  8.         else:
+>  9.             self.id = id
+> 10. 
+> 11.     def get_category(self):
+> 12.         return self.__class__.__name__
+> 13. 
+> 14.     def __str__(self):
+> 15.         return f"An object of type {self.get_category()} with id {self.id}."
+> 16. 
+> 17.     def condition_description(self):
+> 18.         if self.condition > 3:
+> 19.             return "It's great!"
+> 20.         else:
+> 21.             return "It stinks!"
+> ```
+> 
+> **Code Block 2: Clothing Class**
+> ```python
+> 23. from swap_meet.item import Item
+> 24. 
+> 25. class Clothing(Item):
+> 26.     def __init__(self, condition=None, id=None, fabric="Unknown"):
+> 27.         super().__init__(condition=condition, id=id)
+> 28.         self.fabric = fabric
+> 29. 
+> 30.     def __str__(self):
+> 31.         return f"{super()._str_()} It is made from {self.condition} fabric."
+> ```
+>
+> I've included line numbers for reference. If you have any questions or need further clarification, feel free to ask!
+
+</details>  
+
+## Adding ChatGPT to the Debugging Toolbox
+
+The key to effectively using ChatGPT for debugging is that we’re adding a new tool to our toolbox, not replacing the ones we’ve already built. The general steps of our debugging journey should look pretty much the same: we want to determine what is happening, why it’s happening, and how we fix it. 
+
+Let’s take a closer look at the debugging journey to see where we can make the most use of tools like ChatGPT.
+
+### What’s Happening
+
+#### Investigation
+
+When we’re debugging the problem may not always be obvious, so we may have to do some searches about errors, set breakpoints, or add print statements to better understand what kinds of inputs are giving us an issue and how. If our initial searches don’t come up with clear answers, we can ask ChatGPT questions about the error messages or the behavior we are seeing to get a better idea of what is causing the problem and where it may stem from.
```

Comment:
```suggestion
When we’re debugging the problem may not always be obvious, so we may have to do some searches about errors, set breakpoints, or add print statements to better understand what kinds of inputs are giving us an issue and why. If our initial searches don’t come up with clear answers, we can ask ChatGPT questions about the error messages or the behavior we are seeing to get a better idea of what is causing the problem.
```

Reply:
Sentence felt a little long so I feel like the last bit can be cut but not a big deal if you'd like to leave it in.